### ISSUE TITLE
[FEATURE] Add pipeline support in Databricks Workflows

### ISSUE THREAD

**Issue by christinaet** [databricks/terraform-provider-databricks#1926]
### Use-cases
We need to be able to refer to DLT pipelines by name to set up dependencies in Workflows.

### Attempted Solutions
Currently we have to deploy a pipeline first, get the pipeline id, then deploy a complete workflow that has pipeline dependencies. 

### References
The dbx project by Databricks Labs released DLT pipeline support since 0.8.0+. Hopefully the TF provider can catch up soon so we have a single way to manage all Databricks resources.

**Comment by alexott** [comment]
DLT is supported since beginning. If you're creating DLT pipeline with Terraform, then you just need to use `.id` attribute of the `databricks_pipeline` resource. Here is an example: https://github.com/alexott/anomaly_detection_using_databricks/blob/main/terraform/job.tf#L37

If you're creating DLT pipeline outside of terraform, then yes, we need `databricks_pipeline` data source. Is it your case?

**Comment by alexott** [comment]
@christinaet can you please provide what is missing in the current implementation. Otherwise I don't see this as an issue

**Comment by zcking** [comment]
I agree this isn't an issue. I've used the `.id` attribute in workflows combining DLT pipelines with non-DLT pipelines in a single workflow without issue for a while now. And as of #2202 there is the data source as well: https://registry.terraform.io/providers/databricks/databricks/latest/docs/data-sources/pipelines

**Comment by alexott** [comment]
we have solutions to avoid referring by name.

**Comment by litan1106** [comment]
I found the data databricks_pipelines resource is not helpful in real-world scenarios. Ideally, we want to retrieve id and name because just the pipeline id doesn't tell us anything.

data "databricks_pipelines" "all" {}

output "all_pipelines" {
  value = data.databricks_pipelines.all.ids
}

**Comment by alexott** [comment]
Not an issue anymore
